app/src/main/java/com/swent/assos/ui/login/SignUpScreen.kt

```diff
@@ -75,37 +84,37 @@ fun SignUpScreen(navigationActions: NavigationActions) {
             val data = hashMapOf("email" to email)
 
             val functions = FirebaseFunctions.getInstance("europe-west6")
-            // Firebase.functions.useEmulator("10.0.2.2", 5001)
+            val config = Config()
 
-            functions.useEmulator("10.0.2.2", 5001)
             // change the region of the function to europe-west6
-
+            if (config.get_all().contains("functions")) {
+              functions.useEmulator(R.string.emulatorIP.toString(), 5001)
+            }
             functions
                 .getHttpsCallable("oncallFind")
                 .call(data)
                 .addOnSuccessListener { task ->
```

Comment:
you can use _ instead of task because you are not using the task in this lamda, not really needed to change.